### PR TITLE
[docker] separate build target for grpc

### PR DESCRIPTION
**Pull request by rustielin**
### Description

Separate build target for indexer (grpc) and related images. This is to unstrip them, since other CLIs are stripped in the `tools` builder.

### Test Plan

CI builds image on this PR

### CODE REVIEW

**Review by perryjrandall — APPROVED** [review]
Looks correct to me, boss @ibalajiarun to confirm as well?

Also can you be more specific with the test place criteria for success? How can we tell they are unstripped, what is the effect on image size etc

**Comment by rustielin** [comment]
Here's the layers for an image built on this branch and the latest on `main`: 

```
$ docker history --no-trunc   us-docker.pkg.dev/aptos-registry/docker/indexer-grpc:c3ed47d8b46d85889b198a33851c4c3729ca1e53 | grep COPY

<missing>                                                                 About an hour ago   COPY /aptos/dist/aptos-indexer-grpc-post-processor /usr/local/bin/aptos-indexer-grpc-post-processor # buildkit                                                                                      318MB     buildkit.dockerfile.v0
<missing>                                                                 About an hour ago   COPY /aptos/dist/aptos-indexer-grpc-data-service /usr/local/bin/aptos-indexer-grpc-data-service # buildkit                                                                                          429MB     buildkit.dockerfile.v0
<missing>                                                                 About an hour ago   COPY /aptos/dist/aptos-indexer-grpc-file-store /usr/local/bin/aptos-indexer-grpc-file-store # buildkit                                                                                              268MB     buildkit.dockerfile.v0
<missing>                                                                 About an hour ago   COPY /aptos/dist/aptos-indexer-grpc-cache-worker /usr/local/bin/aptos-indexer-grpc-cache-worker # buildkit                                                                                          335MB     buildkit.dockerfile.v0
```

main:
```
$ docker history  --no-trunc  us-docker.pkg.dev/aptos-registry/docker/indexer-grpc:main | grep COPY
<missing>                                                                 2 hours ago   COPY /aptos/dist/aptos-indexer-grpc-post-processor /usr/local/bin/aptos-indexer-grpc-post-processor # buildkit                                                                                                                      10.1MB    buildkit.dockerfile.v0
<missing>                                                                 2 hours ago   COPY /aptos/dist/aptos-indexer-grpc-data-service /usr/local/bin/aptos-indexer-grpc-data-service # buildkit                                                                                                                          10.5MB    buildkit.dockerfile.v0
<missing>                                                                 2 hours ago   COPY /aptos/dist/aptos-indexer-grpc-file-store /usr/local/bin/aptos-indexer-grpc-file-store # buildkit                                                                                                                              8.89MB    buildkit.dockerfile.v0
<missing>                                                                 2 hours ago   COPY /aptos/dist/aptos-indexer-grpc-cache-worker /usr/local/bin/aptos-indexer-grpc-cache-worker # buildkit                                                                                                                          10.4MB    buildkit.dockerfile.v0
```

File: docker/builder/build-indexer.sh
Comment (ibalajiarun) [inline docker/builder/build-indexer.sh:1]: I would consider having a `common.sh` file that abstracts away the common functionality, so we can just call it by passing the list of targets to build.

**Review by ibalajiarun — APPROVED** [review]
This is great because it actually brought down build time down to 9 minutes looks like?

**Pull request merged**